pygeo/parameterization/DVGeo.py

```diff
@@ -2446,6 +2498,38 @@ def addVariablesPyOpt(
                 ("spanwiselocalVars", self.DV_listSpanwiseLocal),
             ]
         )
+
+        # we add the composite DVs, and construct linear constraints that replace the existing bounds
+        # then we simply return without adding any of the other DVs
+        if self.useComposite:
+            dv = self.DVComposite
+            optProb.addVarGroup(dv.name, dv.nVal, "c", value=dv.value, lower=dv.lower, upper=dv.upper, scale=dv.scale)
+
+            # add the linear DV constraints that replace the existing bounds!
+            # Note that we assume all DVs are added here, i.e. no ignoreVars or any of the vars = False
+            lb = {}
+            ub = {}
+            for lst in varLists:
+                for key in varLists[lst]:
+                    dv = varLists[lst][key]
+                    lb[key] = dv.lower
+                    ub[key] = dv.upper
+
+            lb = self.convertDictToSensitivity(lb)
+            ub = self.convertDictToSensitivity(ub)
```

Comment:
I am not sure I understand the trick here, can you elaborate real quick? you just turn the "regular DV" bounds into upper and lower constraint values, and use the singular vector matrix as Jacobian bc this is a linear transformation?
Maybe I am just confused about how pyOptSparse handles linear constraints, but it looks like we are not _evaluating_ them here, right?